### PR TITLE
Fix build failure when run-tests is enabled for rocfft

### PR DESCRIPTION
**Pull request by srekolam**
currently when we do spack install --test=root rocftt, we see a build failure. The PR fixes the build failure.

### CODE REVIEW

File: var/spack/repos/builtin/packages/rocfft/0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+   $<BUILD_INTERFACE:${Boost_INCLUDE_DIRS}>
++  $<BUILD_INTERFACE:${rocrand_INCLUDE_DIRS}>
+   $<BUILD_INTERFACE:${FFTW_INCLUDES}>
+   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/misc/include>
+   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../library/src/include>
+@@ -216,6 +222,7 @@ target_link_libraries( rocfft-test
+   hip::device
+   roc::rocfft
+   hip::hiprand
++  roc::rocrand
```
Comment (cgmb) [inline var/spack/repos/builtin/packages/rocfft/0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch:36]: If this is required, there's something wrong in the hiprand project. Should file a bug upstream.

Comment (srekolam) [reply var/spack/repos/builtin/packages/rocfft/0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch:36]: this is what is being observed -
1 error generated when compiling for gfx1030.
make[2]: *** [clients/tests/CMakeFiles/rocfft-test.dir/build.make:163: clients/tests/CMakeFiles/rocfft-test.dir/accuracy_test_adhoc.cpp.o] Error 1
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/rocfft_accuracy_test.cpp:28:
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/rocfft_accuracy_test.h:24:
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/../../shared/rocfft_params.h:24:
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/../../shared/../shared/fft_params.h:40:
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/../../shared/../shared/data_gen_device.h:36:
In file included from /home/spack/opt/spack/linux-rhel9-skylake_avx512/gcc-11.3.1/hiprand-6.0.2-7mzvsh23dtlwgwzkgqgn2vtgshgkolst/include/hiprand/hiprand.h:66:
/home/spack/opt/spack/linux-rhel9-skylake_avx512/gcc-11.3.1/hiprand-6.0.2-7mzvsh23dtlwgwzkgqgn2vtgshgkolst/include/hiprand/hiprand_rocm.h:24:10: fatal error: 'rocrand/rocrand.h' file not found
#include <rocrand/rocrand.h>
         ^~~~~~~~~~~~~~~~~~~
1 error generated when compiling for gfx1030.
make[2]: *** [clients/tests/CMakeFiles/rocfft-test.dir/build.make:93: clients/tests/CMakeFiles/rocfft-test.dir/rocfft_accuracy_test.cpp.o] Error 1
1 error generated when compiling for gfx1030.
make[2]: *** [clients/tests/CMakeFiles/rocfft-test.dir/build.make:79: clients/tests/CMakeFiles/rocfft-test.dir/gtest_main.cpp.o] Error 1

Comment (cgmb) [reply var/spack/repos/builtin/packages/rocfft/0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch:36]: I filed a bug. https://github.com/ROCm/hipRAND/issues/113

**Comment by srekolam** [comment]
Let me know if this can be merged ?

**Comment by srekolam** [comment]
Can i know if this changes can be merged ?

File: var/spack/repos/builtin/packages/rocfft/package.py [hunk 2]
```diff
@@ -180,6 +177,9 @@ def check(self):
     patch("0003-Fix-clients-fftw3-include-dirs-rocm-4.5.patch", when="@4.5.0:5.1")
     # Patch to add install prefix header location for sqlite for 5.4
     patch("0004-fix-missing-sqlite-include-paths.patch", when="@5.4.0:5.5")
+    patch(
+        "0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch", when="@5.7.0:"
+    )
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/rocfft/package.py:99]: Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

File: var/spack/repos/builtin/packages/rocfft/0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/rocfft/0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch:36]: Do you have an estimate as to when the bug will be fixed upstream? Was wondering if we can wait a bit and retrieve the patch from upstream directly.

**Review by alalazo — DISMISSED** [review]

**Comment by srekolam** [comment]
@tldahlgren can you please let us know if this can be merged ?

**Review by tgamblin — APPROVED** [review]

File: var/spack/repos/builtin/packages/rocfft/0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch (hunk 1 above)
Comment (cgmb) [reply var/spack/repos/builtin/packages/rocfft/0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch:36]: @alalazo. It has been merged for ROCm 6.3. https://github.com/ROCm/hipRAND/commit/a471c3512ed5d16685d42a89fe43f90458abfba0

**Pull request merged**